modules/git/git.go

```diff
@@ -201,6 +201,29 @@ func InitFull(ctx context.Context) (err error) {
 	return syncGitConfig()
 }
 
+func enableReflogs() error {
+	if err := configSet("core.logAllRefUpdates", "true"); err != nil {
+		return err
+	}
+	if setting.Git.Reflog.Expiration != 90 {
```

Comment:
Why not just Update it? Is the check necessary?

Reply:
I figured it was nice to keep the .gitconfig as minimal as possible. It's not necessary, no.

Reply:
I guess it would cause bug if `setting.Git.Reflog.Expiration==90`, but `gc.reflogExpire != 90` in config

Reply:
I think this PR assumes the default value of git reflog expire is always 90?

Reply:
What if a user:

* Set Expiration=14 , write reflogExpire=14 to config
* Change Expiration to 90
* Then the reflogExpire in config is still 14?

Reply:
The config is synced whenever `GlobalInitInstalled` is invoked, via `git.InitFull`. From what I understand, this should happen whenever the server restarts. This setting is therefore like others in `app.ini`, in that they are static for the lifetime of the Gitea process and will be refreshed when the process is restarted.

Reply:
Removed the 90 days check.

Reply:
Yup, I guess the old code might work like this:

* Set Expiration=14 in `app.ini`, start Gitea, write reflogExpire=14 to git config
* Stop Gitea, Change Expiration in `app.ini` to 90
* Start Gitea, `if Expiration != 90` check fails, then the reflogExpire in git config is still 14, it's inconsistent with the value in `app.ini`

I think removing the check is right.

Reply:
Ah I see, so the issue was I wasn't unsetting. Noted, yeah that's what I get for trying to make things simpler, I made it more complicated instead.